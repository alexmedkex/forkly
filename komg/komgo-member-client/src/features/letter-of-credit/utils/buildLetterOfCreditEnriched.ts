import { IDataLetterOfCredit, ILetterOfCredit } from '@komgo/types'
import { findRole } from './findRole'
import { findTasksByLetterOfCreditStaticId } from './taskUtil'
import { TaskStatus, Task } from '../../tasks/store/types'
import { findLatestShipment } from './findLatestShipment'
import { ILetterOfCreditEnriched } from '../store/types'

export const buildLetterOfCreditEnriched = (
  letter: ILetterOfCredit<IDataLetterOfCredit>,
  tasks: Task[],
  companyStaticId: string
) => {
  const role = findRole(letter, companyStaticId)
  const letterTasks = findTasksByLetterOfCreditStaticId(tasks, letter.staticId)
  const enriched: ILetterOfCreditEnriched = {
    ...letter,
    role,
    actionStatus: letterTasks.length ? TaskStatus.ToDo : TaskStatus.Done,
    latestShipment: findLatestShipment(letter),
    tasks: letterTasks
  }
  return enriched
}
