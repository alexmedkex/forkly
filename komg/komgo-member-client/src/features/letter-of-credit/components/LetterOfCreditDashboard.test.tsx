import React from 'react'
import { buildFakeLetterOfCredit, LetterOfCreditTaskType } from '@komgo/types'
import { render, wait, fireEvent, cleanup, act } from '@testing-library/react'
import { MemoryRouter as Router } from 'react-router-dom'
import { fakeTask, mockDate } from '../../letter-of-credit-legacy/utils/faker'
import { LetterOfCreditDashboard, LetterOfCreditDashboardProps } from './LetterOfCreditDashboard'
import { buildLetterOfCreditEnriched } from '../utils/buildLetterOfCreditEnriched'
import { LETTER_OF_CREDIT_TYPE_LABELS } from '../store/types'
import { TaskStatus } from '../../tasks/store/types'

jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children)

describe('LetterOfCreditDashboard', () => {
  const defaultProps: LetterOfCreditDashboardProps = {
    tasks: [],
    standbyLetters: 0,
    documentaryLetters: 0,
    lettersOfCredit: [],
    history: { push: jest.fn() } as any,
    location: {
      pathname: '',
      search: 'tradeId=123',
      state: '',
      hash: ''
    },
    match: {
      isExact: true,
      path: '',
      url: '',
      params: null
    },
    staticContext: undefined
  }
  beforeEach(() => {
    mockDate().freeze('Tue Nov 13 2018 00:00:00 GMT+0000 (UTC)')
    jest.resetAllMocks()
  })

  afterEach(() => {
    afterEach(cleanup)
  })

  it('renders an empty UI', () => {
    const { asFragment } = render(
      <Router>
        <LetterOfCreditDashboard {...defaultProps} />
      </Router>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  describe('Standby', () => {
    it('renders', () => {
      const tasks = []
      const companyStaticId = 'cf63c1f8-1165-4c94-a8f8-9252eb4f0016' // copied from buildLetterOfCreditEnriched
      const offPlatformLetterOfCredit = buildLetterOfCreditEnriched(buildFakeLetterOfCredit(), tasks, companyStaticId)
      offPlatformLetterOfCredit.templateInstance.data.beneficiary.isMember = false
      const lettersOfCredit = [
        buildLetterOfCreditEnriched(buildFakeLetterOfCredit(), tasks, companyStaticId),
        buildLetterOfCreditEnriched(buildFakeLetterOfCredit(), tasks, companyStaticId),
        offPlatformLetterOfCredit
      ]
      const { asFragment } = render(
        <Router>
          <LetterOfCreditDashboard {...defaultProps} lettersOfCredit={lettersOfCredit} />
        </Router>
      )

      expect(asFragment()).toMatchSnapshot()
    })

    it('opens a standby letter of credit', async () => {
      const tasks = []
      const letterStaticId = 'cf5bb000-0e55-43f5-b0f4-34e72201f414'
      const reference = 'ref-123'
      const companyStaticId = 'cf63c1f8-1165-4c94-a8f8-9252eb4f0016' // copied from buildLetterOfCreditEnriched
      const lettersOfCredit = [
        buildLetterOfCreditEnriched(
          buildFakeLetterOfCredit({ staticId: letterStaticId, reference }),
          tasks,
          companyStaticId
        )
      ]
      const component = render(
        <Router>
          <LetterOfCreditDashboard {...defaultProps} lettersOfCredit={lettersOfCredit} />
        </Router>
      )

      await wait(() => {
        fireEvent.click(component.getByTestId(`${reference}-dropdown`))
      })
      await wait(() => {
        fireEvent.click(component.getByText('View SBLC'))
      })

      expect(defaultProps.history.push).toHaveBeenCalledWith(`/letters-of-credit/${letterStaticId}`)
    })

    it('review a standby letter of credit', async () => {
      const letterStaticId = 'cf5bb000-0e55-43f5-b0f4-34e72201f414'
      const reference = 'ref-123'
      const companyStaticId = 'cf63c1f8-1165-4c94-a8f8-9252eb4f0016' // copied from buildLetterOfCreditEnriched
      const task = fakeTask({
        type: LetterOfCreditTaskType.ReviewRequested as any,
        context: { staticId: letterStaticId }
      })
      const lettersOfCredit = [
        buildLetterOfCreditEnriched(
          buildFakeLetterOfCredit({ staticId: letterStaticId, reference }),
          [task],
          companyStaticId
        )
      ]
      const component = render(
        <Router>
          <LetterOfCreditDashboard {...defaultProps} lettersOfCredit={lettersOfCredit} tasks={[task]} />
        </Router>
      )

      await wait(() => {
        fireEvent.click(component.getByTestId(`${reference}-dropdown`))
      })
      await wait(() => {
        fireEvent.click(component.getByText(LETTER_OF_CREDIT_TYPE_LABELS[task.taskType]))
      })

      expect(defaultProps.history.push).toHaveBeenCalledWith(`/letters-of-credit/${letterStaticId}`)
    })

    it('opens a standby letter of credit when a ReviewRequested is done', async () => {
      const letterStaticId = 'cf5bb000-0e55-43f5-b0f4-34e72201f414'
      const reference = 'ref-123'
      const companyStaticId = 'cf63c1f8-1165-4c94-a8f8-9252eb4f0016' // copied from buildLetterOfCreditEnriched
      const task = fakeTask({
        type: LetterOfCreditTaskType.ReviewRequested as any,
        status: TaskStatus.Done,
        context: { staticId: letterStaticId }
      })
      const lettersOfCredit = [
        buildLetterOfCreditEnriched(
          buildFakeLetterOfCredit({ staticId: letterStaticId, reference }),
          [task],
          companyStaticId
        )
      ]
      const component = render(
        <Router>
          <LetterOfCreditDashboard {...defaultProps} lettersOfCredit={lettersOfCredit} tasks={[task]} />
        </Router>
      )

      await wait(() => {
        fireEvent.click(component.getByTestId(`${reference}-dropdown`))
      })
      await wait(() => {
        fireEvent.click(component.getByText('View SBLC'))
      })

      expect(defaultProps.history.push).toHaveBeenCalledWith(`/letters-of-credit/${letterStaticId}`)
    })

    it('opens a standby letter of credit when a ReviewIssued is done', async () => {
      const letterStaticId = 'cf5bb000-0e55-43f5-b0f4-34e72201f414'
      const reference = 'ref-123'
      const companyStaticId = 'cf63c1f8-1165-4c94-a8f8-9252eb4f0016' // copied from buildLetterOfCreditEnriched
      const task = fakeTask({
        type: LetterOfCreditTaskType.ReviewIssued as any,
        status: TaskStatus.Done,
        context: { staticId: letterStaticId }
      })
      const lettersOfCredit = [
        buildLetterOfCreditEnriched(
          buildFakeLetterOfCredit({ staticId: letterStaticId, reference }),
          [task],
          companyStaticId
        )
      ]
      const component = render(
        <Router>
          <LetterOfCreditDashboard {...defaultProps} lettersOfCredit={lettersOfCredit} tasks={[task]} />
        </Router>
      )

      await wait(() => {
        fireEvent.click(component.getByTestId(`${reference}-dropdown`))
      })
      await wait(() => {
        fireEvent.click(component.getByText('View SBLC'))
      })

      expect(defaultProps.history.push).toHaveBeenCalledWith(`/letters-of-credit/${letterStaticId}`)
    })
  })
})
