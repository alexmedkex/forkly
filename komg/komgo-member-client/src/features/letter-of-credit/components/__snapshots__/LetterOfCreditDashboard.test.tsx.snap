// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterOfCreditDashboard Standby renders 1`] = `
<DocumentFragment>
  .c0 {
  padding: 0;
  margin: 0;
  padding-top: 16px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<div
    class="c0"
  >
    <div
      style="display: flex; justify-content: space-between; align-items: flex-start;"
    >
      <h1
        class="ui header"
        style="margin: 0px;"
      >
        Financial Instruments
      </h1>
    </div>
  </div>
  .c0 {
  padding: 0;
  margin: 0;
  padding-left: 24px;
  padding-right: 24px;
  margin-bottom: 24px;
}

<div
    class="c0"
  >
    <a
      aria-current="false"
      href="/letters-of-credit/dashboard/standby"
      style="color: rgb(0, 0, 0); font-weight: 700; margin-right: 16px;"
    >
      Stand by Letters of Credit (0) 
    </a>
  </div>
  .c0 {
  padding: 0;
  margin: 0;
  padding-left: 24px;
  padding-right: 24px;
}

.c1 {
  display: inline-block;
  font-weight: bold;
  width: 120px;
}

.c2 {
  margin: 0;
  line-height: 2;
  white-space: nowrap;
}

.c3.c3.c3 {
  background-color: #009ea8;
  border-color: #009ea8;
  color: white;
}

.c4.c4.c4 {
  background-color: #009ea8;
  border-color: #009ea8;
  color: white;
}

.c5.c5.c5 {
  background-color: #009ea8;
  border-color: #009ea8;
  color: white;
}

<div
    class="c0"
  >
    <div
      class="ReactTable"
    >
      <div
        class="rt-table"
        role="grid"
      >
        <div
          class="rt-thead -header"
          style="min-width: 900px;"
        >
          <div
            class="rt-tr"
            role="row"
          >
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="reference"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Reference
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="issuingBankReference"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Issuing bank reference
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="expiry"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Expiry
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="latestShipment"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Latest shipment / delivery
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="role"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  role
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="amount"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  amount
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="status"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  status
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="action"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  action
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th actions-menu rt-resizable-header"
              data-test-id="actionsMenu"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              />
              <div
                class="rt-resizer"
              />
            </div>
          </div>
        </div>
        <div
          class="rt-tbody"
          style="min-width: 900px;"
        >
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -odd"
              data-test-id="LC-ING-1234"
              role="row"
              style="cursor: pointer;"
            >
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-reference"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    LC-ING-1234
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-issuingBankReference"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    REF-0001
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-expiry"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    2019-11-29
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-latestShipment"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    2018/11/13
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-role"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <div
                    class="ui popup transition visible"
                  >
                    <div
                      class="ui list"
                      role="list"
                      style="width: 300px;"
                    >
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Applicant:
                        </span>
                        <span
                          class="c2"
                        >
                          Applicant Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Beneficiary:
                        </span>
                        <span
                          class="c2"
                        >
                          Beneficiary Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Issuing Bank:
                        </span>
                        <span
                          class="c2"
                        >
                          Issuing Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Advising Bank:
                        </span>
                        Beneficiary Name
                      </div>
                    </div>
                  </div>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-amount"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    USD 10,000.00
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-status"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    Requested
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-action"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span
                    class="ui label c3"
                  >
                    DONE
                  </span>
                </span>
              </div>
              <div
                class="rt-td actions-menu"
                data-test-id="LC-ING-1234-actionsMenu"
                role="gridcell"
                style="overflow: visible; flex: 100 0 auto; width: 100px;"
              >
                <div
                  aria-expanded="false"
                  class="ui dropdown"
                  data-test-id="LC-ING-1234-dropdown"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    aria-live="polite"
                    class="text"
                    role="alert"
                  />
                  <i
                    aria-hidden="true"
                    class="ellipsis horizontal icon"
                  />
                  <div
                    class="left menu transition"
                  >
                    <div
                      class="item"
                      date-test-id="open-1cf0b888-e5ff-47ca-bcf2-b2dc9478b81c"
                      role="option"
                    >
                      View SBLC
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -even"
              data-test-id="LC-ING-1234"
              role="row"
              style="cursor: pointer;"
            >
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-reference"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    LC-ING-1234
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-issuingBankReference"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    REF-0001
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-expiry"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    2019-11-29
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-latestShipment"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    2018/11/13
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-role"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <div
                    class="ui popup transition visible"
                  >
                    <div
                      class="ui list"
                      role="list"
                      style="width: 300px;"
                    >
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Applicant:
                        </span>
                        <span
                          class="c2"
                        >
                          Applicant Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Beneficiary:
                        </span>
                        <span
                          class="c2"
                        >
                          Beneficiary Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Issuing Bank:
                        </span>
                        <span
                          class="c2"
                        >
                          Issuing Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Advising Bank:
                        </span>
                        Beneficiary Name
                      </div>
                    </div>
                  </div>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-amount"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    USD 10,000.00
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-status"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    Requested
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-action"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span
                    class="ui label c4"
                  >
                    DONE
                  </span>
                </span>
              </div>
              <div
                class="rt-td actions-menu"
                data-test-id="LC-ING-1234-actionsMenu"
                role="gridcell"
                style="overflow: visible; flex: 100 0 auto; width: 100px;"
              >
                <div
                  aria-expanded="false"
                  class="ui dropdown"
                  data-test-id="LC-ING-1234-dropdown"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    aria-live="polite"
                    class="text"
                    role="alert"
                  />
                  <i
                    aria-hidden="true"
                    class="ellipsis horizontal icon"
                  />
                  <div
                    class="left menu transition"
                  >
                    <div
                      class="item"
                      date-test-id="open-1cf0b888-e5ff-47ca-bcf2-b2dc9478b81c"
                      role="option"
                    >
                      View SBLC
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rt-tr-group"
            role="rowgroup"
          >
            <div
              class="rt-tr -odd"
              data-test-id="LC-ING-1234"
              role="row"
              style="cursor: pointer;"
            >
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-reference"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    LC-ING-1234
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-issuingBankReference"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    REF-0001
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-expiry"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    -
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-latestShipment"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    2018/11/13
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-role"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <div
                    class="ui popup transition visible"
                  >
                    <div
                      class="ui list"
                      role="list"
                      style="width: 300px;"
                    >
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Applicant:
                        </span>
                        <span
                          class="c2"
                        >
                          Applicant Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Beneficiary:
                        </span>
                        <span
                          class="c2"
                        >
                          Beneficiary Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Issuing Bank:
                        </span>
                        <span
                          class="c2"
                        >
                          Issuing Name
                        </span>
                      </div>
                      <div
                        class="item"
                        role="listitem"
                      >
                        <span
                          class="c1"
                        >
                          Advising Bank:
                        </span>
                        Beneficiary Name
                      </div>
                    </div>
                  </div>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-amount"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    -
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-status"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span>
                    Requested
                  </span>
                </span>
              </div>
              <div
                class="rt-td"
                data-test-id="LC-ING-1234-action"
                role="gridcell"
                style="overflow: hidden; flex: 100 0 auto; width: 100px;"
              >
                <span
                  style="text-align: left; display: block; overflow: hidden; text-overflow: ellipsis;"
                >
                  <span
                    class="ui label c5"
                  >
                    DONE
                  </span>
                </span>
              </div>
              <div
                class="rt-td actions-menu"
                data-test-id="LC-ING-1234-actionsMenu"
                role="gridcell"
                style="overflow: visible; flex: 100 0 auto; width: 100px;"
              >
                <div
                  aria-expanded="false"
                  class="ui dropdown"
                  data-test-id="LC-ING-1234-dropdown"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    aria-live="polite"
                    class="text"
                    role="alert"
                  />
                  <i
                    aria-hidden="true"
                    class="ellipsis horizontal icon"
                  />
                  <div
                    class="left menu transition"
                  >
                    <div
                      class="item"
                      date-test-id="open-1cf0b888-e5ff-47ca-bcf2-b2dc9478b81c"
                      role="option"
                    >
                      View SBLC
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="-loading"
      >
        <div
          class="-loading-inner"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LetterOfCreditDashboard renders an empty UI 1`] = `
<DocumentFragment>
  .c0 {
  padding: 0;
  margin: 0;
  padding-top: 16px;
  padding-bottom: 24px;
  padding-left: 24px;
  padding-right: 24px;
}

<div
    class="c0"
  >
    <div
      style="display: flex; justify-content: space-between; align-items: flex-start;"
    >
      <h1
        class="ui header"
        style="margin: 0px;"
      >
        Financial Instruments
      </h1>
    </div>
  </div>
  .c0 {
  padding: 0;
  margin: 0;
  padding-left: 24px;
  padding-right: 24px;
  margin-bottom: 24px;
}

<div
    class="c0"
  >
    <a
      aria-current="false"
      href="/letters-of-credit/dashboard/standby"
      style="color: rgb(0, 0, 0); font-weight: 700; margin-right: 16px;"
    >
      Stand by Letters of Credit (0) 
    </a>
  </div>
  .c0 {
  padding: 0;
  margin: 0;
  padding-left: 24px;
  padding-right: 24px;
}

<div
    class="c0"
  >
    <div
      class="ReactTable"
    >
      <div
        class="rt-table"
        role="grid"
      >
        <div
          class="rt-thead -header"
          style="min-width: 900px;"
        >
          <div
            class="rt-tr"
            role="row"
          >
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="reference"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Reference
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="issuingBankReference"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Issuing bank reference
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="expiry"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Expiry
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="latestShipment"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  Latest shipment / delivery
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="role"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  role
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="amount"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  amount
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="status"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  status
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th rt-resizable-header -cursor-pointer"
              data-test-id="action"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              >
                <span
                  style="text-align: left; display: block;"
                >
                  action
                </span>
              </div>
              <div
                class="rt-resizer"
              />
            </div>
            <div
              class="rt-th actions-menu rt-resizable-header"
              data-test-id="actionsMenu"
              role="columnheader"
              style="flex: 100 0 auto; width: 100px;"
              tabindex="-1"
            >
              <div
                class="rt-resizable-header-content"
              />
              <div
                class="rt-resizer"
              />
            </div>
          </div>
        </div>
        <div
          class="rt-tbody"
          style="min-width: 900px;"
        />
      </div>
      <div
        class="rt-noData"
      >
        No rows found
      </div>
      <div
        class="-loading"
      >
        <div
          class="-loading-inner"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
